### PR TITLE
neon: implementation of qmovn and helpers

### PR DESCRIPTION
**Pull request by seanptmaher**
Here's an implementation of qmovn.

I looked through some of simde's utility function for something similar to `SIMDE_CONVERT_VECTOR_` that did saturation, but no dice. Did I miss something obvious, or is this good?

Thanks,
Sean

(CC @ngzhian @tlively)

### CODE REVIEW

**Comment by nemequ** [comment]
> I looked through some of simde's utility function for something similar to SIMDE_CONVERT_VECTOR_ that did saturation, but no dice. Did I miss something obvious, or is this good?

Nope, you're not missing anything.  The patch looks good.

Creating optimized implementations of these is going to be interesting; most architectures have support for saturating + narrowing, but they don't support 64-bit vectors; instead, they generally take 2 128-bit inputs and produce one 128-bit output.  I have a feeling our best option will be to just do something like `_mm_packs_epi32(a, a)` (on x86, or `vec_packs(a, a)` on AltiVec, etc.) and just ignore the upper (or lower) half.

I haven't really been mentioning the possible implementations for other architectures, but is that something you'd be interested in?  If you want I can go through the NEON implementations already in SIMDe and point you to the right functions for at least some of them…

**Comment by seanptmaher** [comment]
Right now, I'm mostly working on trying to get a specific program (TensorFlow Lite) compiling using SIMDe, I wanted to take a look at what kinds of performance it'd give without native implementations, and then follow up by adding in as many native implementations as I could, and I'm not yet done my Long List of Intrinsics:tm:, so I can't promise to be able to spend time on revisiting it, yet, but I'm expecting to go back and review the egregious ones soon.

File: test/arm/neon/qmovn.c [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    { {  INT16_C( 18641),  INT16_C( 15455),  INT16_C(  6783), -INT16_C(     7),  INT16_C( 10919),  INT16_C(   129),  INT16_C( 27850),  INT16_C( 28360) },
+      {      INT8_MAX,      INT8_MAX,      INT8_MAX, -INT8_C(   7),      INT8_MAX,      INT8_MAX,      INT8_MAX,      INT8_MAX } },
+    { { -INT16_C(  2336),  INT16_C(  8979),  INT16_C( 22268), -INT16_C( 26471), -INT16_C( 26228), -INT16_C( 10458), -INT16_C(  5364), -INT16_C(  8722) },
+      {      INT8_MIN,      INT8_MAX,      INT8_MAX,      INT8_MIN,      INT8_MIN,      INT8_MIN,      INT8_MIN,      INT8_MIN } },
+
+  };
+
+  for (size_t i = 0 ; i < (sizeof(test_vec) / sizeof(test_vec[0])) ; i++) {
+    simde_int16x8_t a = simde_vld1q_s16(test_vec[i].a);
+    simde_int8x8_t r = simde_vqmovn_s16(a);
```
[the remark this answers is not in the document]
Comment (seanptmaher) [reply test/arm/neon/qmovn.c:34]: Sounds good, I'll add that in.

**Comment by nemequ** [comment]
Merged, thanks!

**Pull request closed without merging**